sound/soc/sof/topology.c

```diff
@@ -1865,8 +1996,15 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 	case snd_soc_dapm_siggen:
 		ret = sof_widget_load_siggen(scomp, index, swidget, tw, &reply);
 		break;
+	/*
+	 * snd_soc_dapm_sink can treated as a special processing type widget.
+	 * So, re-use the same ipc structure and tokens as the effect widgets.
+	 */
+	case snd_soc_dapm_sink:
+		/* fallthrough */
```

Comment:
Could this be conflict with the possible future use of the normal dapm_sink widget?